src/realm/CMakeLists.txt

```diff
@@ -261,65 +261,21 @@ if(NOT MSVC)
     list(APPEND REALM_SOURCES util/interprocess_mutex.cpp)
 endif()
 
-# We add the headers to the library only so they show up in dev
-# environments. It won't actually affect compilation process.
-add_library(CoreObjects OBJECT ${REALM_SOURCES} ${UTIL_SOURCES} ${REALM_INSTALL_ALL_HEADERS})
-set_target_properties(CoreObjects PROPERTIES
-    POSITION_INDEPENDENT_CODE ON
+add_library(Storage STATIC
+    ${REALM_SOURCES}
+    ${UTIL_SOURCES}
+    ${REALM_INSTALL_ALL_HEADERS}
+    ${REALM_OBJECT_FILES}
+    $<TARGET_OBJECTS:Bid>
 )
-target_compile_definitions(CoreObjects PRIVATE
-  "$<$<CXX_COMPILER_ID:MSVC>:_CRT_SECURE_NO_WARNINGS>"
-  "$<$<CXX_COMPILER_ID:MSVC>:_SCL_SECURE_NO_WARNINGS>"
-)
-target_compile_definitions(CoreObjects PRIVATE
-  $<$<CONFIG:Debug>:REALM_DEBUG=1>
-  $<$<CONFIG:MinSizeDebug>:REALM_DEBUG=1>
-)
-target_compile_options(CoreObjects PRIVATE ${REALM_SANITIZER_FLAGS})
-
-set(REALM_OBJECT_FILES $<TARGET_OBJECTS:CoreObjects>)
-list(APPEND REALM_OBJECT_FILES $<TARGET_OBJECTS:Bid>)
-
-if(COMMAND set_target_xcode_attributes)
-    set_target_xcode_attributes(CoreObjects)
-endif()
-
-if(CMAKE_SYSTEM_NAME MATCHES "^Windows")
-    add_dependencies(CoreObjects sha_win32 getopt_win32)
-    list(APPEND REALM_OBJECT_FILES $<TARGET_OBJECTS:sha_win32>)
-    if(CMAKE_SYSTEM_NAME STREQUAL "Windows")
-        list(APPEND REALM_EXTRA_LIBS Version.lib psapi.lib)
-    endif()
-elseif(ANDROID)
-    list(APPEND REALM_EXTRA_LIBS atomic)
-endif()
+add_library(Realm::Storage ALIAS Storage)
 
-set(REALM_XCODE_DEPENDENCY "")
-if(CMAKE_GENERATOR STREQUAL Xcode)
-    # The placeholder file is needed in order to make builds work on Xcode
-    # The issue is described in the last sentence here:
-    # https://cmake.org/cmake/help/v3.7/command/add_library.html#object-libraries
-    # Another problem is that Xcode does not see changes in the CoreObjects
-    # target, to solve this we trigger rebuiliding the library by making
-    # placeholder.cpp depend on all the sources rather than CoreObjects.
-    set(REALM_XCODE_DEPENDENCY "placeholder.cpp")
```

Comment:
Really happy to see this is not a problem anymore. I'm not sure when it was fixed, but I checked with my Xcode project and it is tracking changes properly.

Reply:
I'm not sure if it's still a problem or not, but we're not using an `OBJECT` target to build `Storage` anymore, so the workaround for tracking changes in `OBJECT` targets in Xcode is no longer needed.